addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java

```diff
@@ -143,49 +175,132 @@ private void changeState(final InternalLedState newState) throws IOException {
         }
     }
 
-    private void fadeToState(final InternalLedState newState) throws IOException {
-        if (!newState.equals(realTargetState)) {
-            keepFading = false;
-            realTargetState = newState;
-
-            executorService.schedule(() -> {
-                if (currentTargetState.equals(newState)) {
-                    return;
-                }
-
-                keepFading = true;
-
-                try (Socket socket = new Socket(host, port)) {
-                    logger.debug("Connected to '{}'", socket);
-
-                    socket.setSoTimeout(DEFAULT_SOCKET_TIMEOUT);
-
-                    try (DataOutputStream outputStream = new DataOutputStream(socket.getOutputStream())) {
-                        // ensure controller is on
-                        sendRaw(getBytesForPower(true), outputStream);
+    /**
+     * Runnable that takes care of fading of the LED's
+     */
+    final static class LEDFaderRunner implements Runnable {
+        protected Logger logger = LoggerFactory.getLogger(LEDFaderRunner.class);
+        protected final int steps;
+
+        // Keep track of LED states
+        protected int fadeStepper = 1;
+        protected InternalLedState currentFadeState;
+        protected InternalLedState toState;
+        protected InternalLedState fromState;
+
+        // After the last update keep the communication channel open for XXX ms
+        protected long keepCommOpenForMS = 1000;
```

Comment:
1) This one is always overridden from the constructor maybe leave its value away. 
2) Better choose a longer name so you don't need the documentation then it is also clear on the place where you use the field.
3) imho code becomes way more readable if order the fields in the same order as the assignments in the constructor and the order in which the parameters are passed. This makes it easy to spot how the fields are used and if there are any inconsistencies.